### DISCUSSION TITLE
Preferences vs EEPROM

### DISCUSSION THREAD

**Discussion by puzzlelogic** [davetcc/tcMenu#529] [Q&A]
Hi, I have just found tcMenu and am looking to start trying it out on a project that is already underway.  I currently use preferences.h (ESP32 C3) to store variables in NVS.  I was hoping I could turn off all the EEPROM functions and save the variables I need manually (I note that there is work underway to add preferences support but I would like to get going before that work matures).

Being new to tcMenu I am not sure what settings on the designer to turn off/on to stop all EEPROM saving.

Can someone help me out?
Jez

**Comment by davetcc** [comment]
We do plan to provide support for preferences api but it needs a bit of thought how to implement. The eeprom locations set in designer don’t do anything directly unless you both select an EEPROM in code generator and call load or save yourself.

**Comment by davetcc** [comment]
I'll close this now. The preferences API is supported in 4.5 and can even be added from web designer. I'll close this now.
